pre_commit/clientlib/validate_config.py

```diff
@@ -53,6 +53,9 @@ def try_regex(repo, hook, value, field_name):
 
 def validate_config_extra(config):
     for repo in config:
+        if 'sha' not in repo:
+            raise InvalidConfigError('Missing "sha" field'
+                                     'for repository {0}'.format(repo['repo']))
```

Comment:
minor nitpick (sorry!):

The predominant style in this repository is to do function calls:

```
like_this(
    with_args,
    here,
)
```

Mostly because it slightly reduces merge conflicts when adding arguments etc.

I can just fix this up after merging if you don't want to bother though :)